programs/mango-v4/src/lib.rs

```diff
@@ -1624,88 +1686,119 @@ pub mod mango_v4 {
         ctx: Context<OpenbookV2RegisterMarket>,
         market_index: OpenbookV2MarketIndex,
         name: String,
+        oracle_price_band: f32,
     ) -> Result<()> {
+        #[cfg(feature = "enable-gpl")]
+        instructions::openbook_v2_register_market(ctx, market_index, name, oracle_price_band)?;
         Ok(())
     }
 
     pub fn openbook_v2_edit_market(
         ctx: Context<OpenbookV2EditMarket>,
         reduce_only_opt: Option<bool>,
         force_close_opt: Option<bool>,
+        name_opt: Option<String>,
     ) -> Result<()> {
+        #[cfg(feature = "enable-gpl")]
+        instructions::openbook_v2_edit_market(ctx, reduce_only_opt, force_close_opt, name_opt)?;
         Ok(())
     }
 
     pub fn openbook_v2_deregister_market(ctx: Context<OpenbookV2DeregisterMarket>) -> Result<()> {
+        #[cfg(feature = "enable-gpl")]
+        instructions::openbook_v2_deregister_market(ctx)?;
         Ok(())
     }
 
-    pub fn openbook_v2_create_open_orders(
-        ctx: Context<OpenbookV2CreateOpenOrders>,
-        account_num: u32,
-    ) -> Result<()> {
+    pub fn openbook_v2_create_open_orders(ctx: Context<OpenbookV2CreateOpenOrders>) -> Result<()> {
+        #[cfg(feature = "enable-gpl")]
+        instructions::openbook_v2_create_open_orders(ctx)?;
         Ok(())
     }
 
     pub fn openbook_v2_close_open_orders(ctx: Context<OpenbookV2CloseOpenOrders>) -> Result<()> {
+        #[cfg(feature = "enable-gpl")]
+        instructions::openbook_v2_close_open_orders(ctx)?;
         Ok(())
     }
 
     #[allow(clippy::too_many_arguments)]
     pub fn openbook_v2_place_order(
         ctx: Context<OpenbookV2PlaceOrder>,
-        side: u8, // openbook_v2::state::Side
-        limit_price: u64,
-        max_base_qty: u64,
-        max_native_quote_qty_including_fees: u64,
-        self_trade_behavior: u8, // openbook_v2::state::SelfTradeBehavior
-        order_type: u8,          // openbook_v2::state::PlaceOrderType
+        side: openbook_v2::state::Side,
+        price_lots: i64,
+        max_base_lots: i64,
+        max_quote_lots_including_fees: i64,
         client_order_id: u64,
+        order_type: openbook_v2::state::PlaceOrderType,
```

Comment:
need to double check that anchor can deal with these foreign enums when it generates the idl

Reply:
it cannot! re-created those enums that appear in the interface, in our codebase to fix this